plugins/browser-plugin-vimeo-tracking/src/mediaPluginBinding.ts

```diff
@@ -0,0 +1,215 @@
+      break;
+
+    case InternalVimeoEvent.PAUSE:
+      trackMediaPause({ id });
+      break;
+
+    case InternalVimeoEvent.ENDED:
+      trackMediaEnd({ id });
+      break;
+
+    case InternalVimeoEvent.SEEKING:
+      trackMediaSeekStart({ id });
+      break;
+
+    case InternalVimeoEvent.SEEKED:
+      trackMediaSeekEnd({ id });
+      break;
+
+    case InternalVimeoEvent.ERROR:
+      trackError(eventData as VimeoError);
+      break;
+
+    case InternalVimeoEvent.TIMEUPDATE:
+      updateMediaTracking({ id, player: { currentTime: (eventData as TimeUpdateData).seconds } });
+      break;
+
+    case InternalVimeoEvent.BUFFERSTART:
+      trackMediaBufferStart({ id });
+      break;
+
+    case InternalVimeoEvent.BUFFEREND:
+      trackMediaBufferEnd({ id });
+      break;
+
+    case InternalVimeoEvent.VOLUMECHANGE:
+      trackedPlayer.player
+        .getVolume()
+        .then((newVolume) => trackMediaVolumeChange({ id, newVolume: Math.round(newVolume * 100) }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.FULLSCREENCHANGE:
+      trackedPlayer.player
+        .getFullscreen()
+        .then((fullscreen) => trackMediaFullscreenChange({ id, fullscreen }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.PLAYBACKRATECHANGE:
+      trackedPlayer.player
+        .getPlaybackRate()
+        .then((newRate) => trackMediaPlaybackRateChange({ id, newRate }))
+        .catch(trackError);
+      break;
+
+    case InternalVimeoEvent.ENTERPICTUREINPICTURE:
+      trackMediaPictureInPictureChange({ id, pictureInPicture: true });
+      break;
+
+    case InternalVimeoEvent.LEAVEPICTUREINPICTURE:
+      trackMediaPictureInPictureChange({ id, pictureInPicture: false });
+      break;
+
+    case InternalVimeoEvent.QUALITYCHANGE:
+      trackMediaQualityChange({ id, newQuality: (eventData as QualityChangeEvent).quality });
+      break;
+
+    case InternalVimeoEvent.CUEPOINT:
+      // `time` can be a keyword in some warehouses, so we rename to `cuePointTime`
+      (eventData as VimeoPlayerCuePointData).cuePointTime = (eventData as VimeoCuePointData).time;
+      delete (eventData as VimeoCuePointData).time;
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.CUEPOINT,
+          data: eventData as VimeoPlayerCuePointData,
+        },
+      });
+      break;
+
+    case InternalVimeoEvent.TEXTTRACKCHANGE:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.TEXT_TRACK_CHANGE,
+          data: eventData as TextTrackData,
+        },
+      });
+      break;
+
+    case InternalVimeoEvent.INTERACTIVEHOTSPOTCLICKED:
+      trackMediaSelfDescribingEvent({
+        id,
+        event: {
+          schema: VimeoSchema.HOTSPOT_CLICK,
+          data: {},
+        },
+        context: [{ schema: VimeoSchema.INTERACTION, data: eventData as VimeoInteractiveHotspotClickedData }],
```

Comment:
I am a bit scared by passing the values directly to the event or entity without filtering for only the known fields – in case the Vimeo APIs change and add a new property, it will result in validation errors.

Reply:
Added explicit conversion functions rather than `as` casts.